core/src/banking_stage.rs

```diff
@@ -890,6 +897,21 @@ impl BankingStage {
                     tx_results.rent_debits,
                 );
             }
+
+            // track committed transactions' cost
+            let transactions: Vec<_> = batch.transactions_iter().cloned().collect();
+            let execution_results = results.to_vec();
```

Comment:
Can this be offloaded to the separate thread as well?

Reply:
yea, I'd love to but feel sketch of sending ref of `batch` and `results` via channel to another thread. I probably can move `results` to sender since it is the last one used, but not `batch`. Not sure. what is the best practice.